code/modules/mining/lavaland/loot/legion_loot.dm

```diff
@@ -39,16 +50,81 @@
 			"<span class='notice'>You hold [src] skyward, dispelling the storm!</span>")
 			playsound(user, 'sound/magic/staff_change.ogg', 200, 0)
 			A.wind_down()
-			return
-	else
-		A = new storm_type(list(user_turf.z))
-		A.name = "staff storm"
-		A.area_type = user_area.type
-		A.telegraph_duration = 100
-		A.end_duration = 100
-
-	user.visible_message("<span class='warning'>[user] holds [src] skywards as red lightning crackles into the sky!</span>", \
-	"<span class='notice'>You hold [src] skyward, calling down a terrible storm!</span>")
-	playsound(user, 'sound/magic/staff_change.ogg', 200, 0)
-	A.telegraph()
-	storm_cooldown = world.time + 200
+			var/old_color = user.color
+			user.color = list(340/255, 240/255, 0,0, 0,0,0,0, 0,0,0,0, 0,0,0,1, 0,0,0,0)
+			var/old_transform = user.transform
+			user.transform *= 1.2
+			animate(user, color = old_color, transform = old_transform, time = 1 SECONDS)
+
+/obj/item/storm_staff/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!thunder_charges)
+		to_chat(user, "<span class='warning'>The staff needs to recharge.</span>")
+		return
+	var/turf/target_turf = get_turf(target)
+	var/area/target_area = get_area(target)
+	var/area/user_area = get_area(user)
+	if(!target_turf || !target_area || (is_type_in_list(target_area, excluded_areas)) || !user_area || (is_type_in_list(user_area, excluded_areas)))
+		to_chat(user, "<span class='warning'>The staff will not work here.</span>")
+		return
+	if(target_turf in targeted_turfs)
+		to_chat(user, "<span class='warning'>That turf is already being shocked!</span>")
```

Comment:
Saying `turf` in a message to players feels a bit iffy, can you reword this, perhaps to something like this?
```suggestion
		to_chat(user, "<span class='warning'>That spot is already being shocked!</span>")
```

Reply:
done